boards/PDM/Src/App/App_Efuse.c

```diff
@@ -1,17 +1,209 @@
+    assert(efuse != NULL);
+
+    efuse->io_efuse              = io_efuse;
+    efuse->enable_channel_0      = enable_channel_0;
+    efuse->disable_channel_0     = disable_channel_0;
+    efuse->enable_channel_1      = enable_channel_1;
+    efuse->disable_channel_1     = disable_channel_1;
+    efuse->is_channel_0_enabled  = is_channel_0_enabled;
+    efuse->is_channel_1_enabled  = is_channel_1_enabled;
+    efuse->stdby_reset           = stdby_reset;
+    efuse->get_channel_0_current = get_channel_0_current;
+    efuse->get_channel_1_current = get_channel_1_current;
+    efuse->channel_0_min_current = channel_0_min_current;
+    efuse->channel_0_max_current = channel_0_max_current;
+    efuse->channel_1_min_current = channel_1_min_current;
+    efuse->channel_1_max_current = channel_1_max_current;
+
+    efuse->num_attempts_channel_0      = 0;
+    efuse->num_attempts_channel_1      = 0;
+    efuse->has_timer_started_channel_0 = false;
+    efuse->has_timer_started_channel_1 = false;
+
+    App_Timer_InitTimer(&efuse->efuse_timer_channel_0, 1000);
+    App_Timer_InitTimer(&efuse->efuse_timer_channel_1, 1000);
+
+    return efuse;
+}
+
+void App_Efuse_Destroy(struct Efuse *efuse)
+{
+    free(efuse);
+}
+
+void App_Efuse_EnableChannel0(struct Efuse *efuse)
+{
+    efuse->enable_channel_0(efuse->io_efuse);
+}
+
+void App_Efuse_DisableChannel0(struct Efuse *efuse)
+{
+    efuse->disable_channel_0(efuse->io_efuse);
+}
+
+void App_Efuse_EnableChannel1(struct Efuse *efuse)
+{
+    efuse->enable_channel_1(efuse->io_efuse);
+}
+
+void App_Efuse_DisableChannel1(struct Efuse *efuse)
+{
+    efuse->disable_channel_1(efuse->io_efuse);
+}
+
+bool App_Efuse_IsChannel0Enabled(struct Efuse *efuse)
+{
+    return efuse->is_channel_0_enabled(efuse->io_efuse);
+}
+
+bool App_Efuse_IsChannel1Enabled(struct Efuse *efuse)
+{
+    return efuse->is_channel_1_enabled(efuse->io_efuse);
+}
+
+void App_Efuse_StandbyReset(struct Efuse *efuse)
+{
+    efuse->stdby_reset(efuse->io_efuse);
+}
+
+float App_Efuse_GetChannel0Current(struct Efuse *efuse)
+{
+    return efuse->get_channel_0_current(efuse->io_efuse);
+}
+
+float App_Efuse_GetChannel1Current(struct Efuse *efuse)
+{
+    return efuse->get_channel_1_current(efuse->io_efuse);
+}
+
+bool App_Efuse_Channel0_CurrentLowCheck(struct Efuse *efuse)
+{
+    return efuse->get_channel_0_current(efuse->io_efuse) < efuse->channel_0_min_current;
+}
+
+bool App_Efuse_Channel0_CurrentHighCheck(struct Efuse *efuse)
+{
+    return efuse->get_channel_0_current(efuse->io_efuse) > efuse->channel_0_max_current;
+}
+
+bool App_Efuse_Channel1_CurrentLowCheck(struct Efuse *efuse)
+{
+    return efuse->get_channel_1_current(efuse->io_efuse) < efuse->channel_1_min_current;
+}
+
+bool App_Efuse_Channel1_CurrentHighCheck(struct Efuse *efuse)
+{
+    return efuse->get_channel_1_current(efuse->io_efuse) > efuse->channel_1_max_current;
+}
+
+int App_Efuse_FaultProcedureChannel0(struct Efuse *efuse, int max_attempts)
```

Comment:
Imo max attempts should not be a parameter here, since it never changes for a single efuse. Consider passing it in during App_Efuse_Create and save it in the efuse struct